core/src/main/java/org/elasticsearch/search/aggregations/bucket/missing/MissingParser.java

```diff
@@ -18,31 +18,24 @@
  */
 package org.elasticsearch.search.aggregations.bucket.missing;
 
-import org.elasticsearch.common.ParseField;
-import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
 import org.elasticsearch.search.aggregations.support.AbstractValuesSourceParser.AnyValuesSourceParser;
-import org.elasticsearch.search.aggregations.support.XContentParseContext;
-import org.elasticsearch.search.aggregations.support.ValueType;
-import org.elasticsearch.search.aggregations.support.ValuesSourceType;
 
 import java.io.IOException;
-import java.util.Map;
 
 public class MissingParser extends AnyValuesSourceParser {
 
-    public MissingParser() {
-        super(true, true);
-    }
+    private final ObjectParser<MissingAggregationBuilder, QueryParseContext> parser;
 
-    @Override
-    protected boolean token(String aggregationName, String currentFieldName, XContentParser.Token token,
-                            XContentParseContext context, Map<ParseField, Object> otherOptions) throws IOException {
-        return false;
+    public MissingParser() {
+        parser = new ObjectParser<>(MissingAggregationBuilder.NAME);
```

Comment:
I think this one is a prime argument for my "this can be a static method/field on the builder" argument.